### PR TITLE
Integrate onednnv2.7-pc2

### PR DESCRIPTION
**Pull request by hyunback**
Integrate oneDNNv2.7pc2 into openVINO GPU plugin

### Tickets:
 - *82062*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_weights.cl [hunk 1]
```diff
@@ -443,6 +443,30 @@ inline uint get_is_os_zyx_isa8_osv8_isv2_index(uint o, uint i, uint z, uint y, u
...
+                 isv1_idx * 8 * 4 +
+                 x * 8 * 8 * 4 +
+                 y * size_x * 8 * 8 * 4 +
+                 z * size_y * size_x * 8 * 8 * 4 +
+                 os_idx * size_z * size_y * size_x * 8 * 8 * 4 +
+                 is_idx * of_8_aligned * size_z * size_y * size_x * 8 * 8 * 4;
+
+    return idx;
+}
+
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_weights.cl:469]: Let's merge `get_is_os_zyx_isa8_osv8_isv4_index` with `isv2` case.

File: src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_weights.cl [hunk 2]
```diff
@@ -499,29 +523,57 @@ inline uint get_g_os_is_zyx_isa8_osv8_isv2_index(uint g, uint o, uint i, uint z,
...
+        o, i, z, y, x,                                                   \
+        CAT(prefix, _SIZE_X),                                            \
+        CAT(prefix, _SIZE_Y),                                            \
+        CAT(prefix, _SIZE_Z),                                            \
+        CAT(prefix, _IFM_NUM),                                           \
+        CAT(prefix, _OFM_NUM),                                           \
+        CAT(prefix, _OFFSET))
+
+inline uint get_g_os_is_yx_isa8_osv8_isv4_index(uint g, uint o, uint i, uint y, uint x, uint size_x,
+                                                uint size_y, uint size_ifm, uint size_ofm, uint offset)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/cl_kernels/include/batch_headers/fetch_weights.cl:537]: Let's merge this with `get_g_os_is_zyx_isa8_osv8_isv2_index`

File: src/plugins/intel_gpu/src/kernel_selector/core/common/kernel_selector_utils.cpp [hunk 3]
```diff
@@ -393,13 +403,7 @@ std::vector<size_t> GetOptimalLocalWorkGroupSizes(std::vector<size_t> gws, const
...
         while (gws[priority_order[i]] % lws_values[lws_idx]) lws_idx++;
 
-        if (lws_max == 256 || total_lws == total_gws) {
-            lws[priority_order[i]] = lws_values[lws_idx];
-        } else {
-            lws[priority_order[i]] = i == 2 && gws[priority_order[0]] != 1 ? 1 : lws_values[lws_idx];
-            if (total_gws > 100 && total_lws < 8 && i == 2)
-                lws[priority_order[i]] = lws_values[lws_idx];
-        }
+        lws[priority_order[i]] = lws_values[lws_idx];
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/core/common/kernel_selector_utils.cpp:406]: I'm afraid this might have impact on prev-gen HWs...

Comment (hyunback) [reply src/plugins/intel_gpu/src/kernel_selector/core/common/kernel_selector_utils.cpp]: I add perf drop defense code for dg2. Other platforms have the same behavior as before.

**Review by isanghao — APPROVED** [review]

**Pull request merged**